test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+      await expect(cardPaymentProcessor.connect(deployer).revokePayment(
+        authorizationId,
+        reversingPaymentCorrelationId,
+        parentTxHash
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_ACCOUNT_IS_NOT_WHITELISTED);
+    });
+
+    it("Is reverted if the payment authorization ID is zero", async () => {
+      await expect(cardPaymentProcessor.connect(admin).revokePayment(
+        ZERO_BYTES16_STRING,
+        reversingPaymentCorrelationId,
+        parentTxHash
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO);
+    });
+
+    it("Is reverted if the parent transaction hash is zero", async () => {
+      await expect(cardPaymentProcessor.connect(admin).revokePayment(
+        authorizationId,
+        reversingPaymentCorrelationId,
+        ZERO_BYTES32_STRING,
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PARENT_TX_HASH_IS_ZERO);
+    });
+
+    it("Is reverted if the payment with the provided authorization ID does not exist", async () => {
+      await expect(cardPaymentProcessor.connect(admin).revokePayment(
+        createBytesString(payment.authorizationId + 1, BYTES16_LENGTH),
+        reversingPaymentCorrelationId,
+        parentTxHash
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST);
+    });
+
+    describe("Transfers tokens as expected, emits the correct event, changes the state properly", async () => {
+      const expectedClearedBalance: number = 0
+      const expectedUnclearedBalance: number = 0;
+      const expectedRevocationCounter: number = 1;
+
+      async function checkRevocation(wasPaymentCleared: boolean) {
+        await checkCardPaymentProcessorState([payment]);
+        await expect(cardPaymentProcessor.connect(admin).revokePayment(
+          authorizationId,
+          reversingPaymentCorrelationId,
+          parentTxHash
+        )).to.changeTokenBalances(
+          brlcMock,
+          [cardPaymentProcessor, payment.account],
+          [-payment.account, +payment.amount]
+        ).and.to.emit(
+          cardPaymentProcessor,
+          "RevokePayment"
+        ).withArgs(
+          authorizationId,
+          reversingPaymentCorrelationId,
+          payment.account.address,
+          payment.amount,
+          expectedClearedBalance,
+          expectedUnclearedBalance,
+          wasPaymentCleared,
+          parentTxHash,
+          expectedRevocationCounter
+        );
+        payment.status = PaymentStatus.Revoked;
+        payment.revocationCounter = expectedRevocationCounter;
+        await checkCardPaymentProcessorState([payment]);
+      }
+
+      it("If payment is uncleared", async () => {
+        const wasPaymentCleared: boolean = false;
+        await checkRevocation(wasPaymentCleared);
+      });
+
+      it("If payment is cleared", async () => {
+        const wasPaymentCleared: boolean = true;
+        await clearPayments([payment], admin);
+        await checkRevocation(wasPaymentCleared);
+      });
+    })
```

Comment:
Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
    });
```